src/librustc/ty/query/plumbing.rs

```diff
@@ -540,7 +540,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
             p.record_query(Q::CATEGORY);
         });
 
-        let res = job.start(self, |tcx| {
+        let res = ::middle::recursion_limit::ensure_sufficient_stack(|| job.start(self, |tcx| {
```

Comment:
Could the `ensure_sufficient_stack` call be moved into the `Job::start` method?

Reply:
Yes, that would probably be a good idea.